### PR TITLE
internal: Remove network access from Code extension

### PR DESCRIPTION
**Pull request by lnicola**

### CODE REVIEW

File: docs/user/manual.adoc [hunk 1]
```diff
@@ -653,10 +653,12 @@ Here is a **non-exhaustive** list of ways to make rust-analyzer execute arbitrar
 The LSP server performs no network access in itself, but runs `cargo metadata` which will update or download the crate registry and the source code of the project dependencies.
 If enabled (the default), build scripts and procedural macros can do anything.
 
-The Code extension automatically connects to GitHub to download updated LSP binaries and, if the nightly channel is selected, to perform update checks using the GitHub API. For `rust-analyzer` developers, using `cargo xtask release` uses the same API to put together the release notes.
+The Code extension does access the network.
```
Comment (bjorn3) [inline docs/user/manual.adoc:656]: *not

File: editors/code/src/main.ts [hunk 2]
```diff
@@ -328,85 +227,39 @@ async function getServer(config: Config, state: PersistentState): Promise<string
...
+        await state.updateServerVersion(config.package.version);
+        return server.fsPath;
     }
 
-    await state.updateServerVersion(config.package.version);
-    return dest.fsPath;
+    await state.updateServerVersion(undefined);
+    await vscode.window.showErrorMessage(
+        "Unfortunately we don't ship binaries for your platform yet. " +
+        "You need to manually clone rust-analyzer repository and " +
```
Comment (bjorn3) [inline editors/code/src/main.ts:256]:
```suggestion
        "You need to manually clone the rust-analyzer repository and " +
```

File: editors/code/src/main.ts [hunk 3]
```diff
@@ -328,85 +227,39 @@ async function getServer(config: Config, state: PersistentState): Promise<string
...
-            progressTitle: "Downloading rust-analyzer server",
-            gunzip: true,
-            mode: 0o755,
-            proxySettings: config.proxySettings,
-        });
-    });
-
-    // Patching executable if that's NixOS.
-    if (await isNixOs()) {
-        await patchelf(dest);
```
Comment (bjorn3) [inline editors/code/src/main.ts:405]: Is patchelf necessary even with a bundled language server on nixos?

Comment (lnicola) [reply editors/code/src/main.ts:405]: I assume it's the same as with a downloaded language server and it won't find `libc` or whatever.

**Pull request merged**